implementation/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTTokenParser.java

```diff
@@ -86,6 +97,10 @@ public JwtContext parse(final String token, final JWTAuthContextInfo authContext
             JwtClaims claimsSet = jwtContext.getJwtClaims();
 
             claimsSet.setClaim(Claims.raw_token.name(), token);
+            if (authContextInfo.getDefaultSubjectClaim() != null) {
+                claimsSet.setClaim(Claims.sub.name(),
+                        claimsSet.getClaimValue(authContextInfo.getDefaultSubjectClaim()));
+            }
```

Comment:
What I was suggesting earlier is to fail at this point if after all the checks `sub` is still null because my understanding is MP JWT, if it accepts the token. should be able to offer a valid (named) `Principal`, so that if `upn` or `preferred_username` are also missing then `sub` can be used as a last resource.
@starksm64, what do you think ?

Reply:
Well, if we are using the default combination of sub and setRequireSubjectm jose4j will validate that for us. If that is not the case, then maybe we should implement a custom Validation so we do the additional checks on `ups` and `preferred_username` and let jose4j validate that as well.

Reply:
@radcortez or may be we can reintroduce your suggested required.subject property and fail now only if sub is still null and the user has explicitly said it is not really needed, just need to check with @starksm64

Reply:
Ok. Cool. If that is the case, then we only need to play with setRequireSubject in jose4j

Reply:
i'd like to keep it simple, the early `sub` validation is a problem in your case, so this PR removes it, and tries to get it from some other custom claim (the user may set it to `sub` but we are not really doing this custom claim check just to delay the default `sub` validation), but in the end we need to either fail if the `sub` (possibly mapped from the custom claim) is still null or let it go if the users allows for it. 
Now that I think about it, it is really a bigger issue, whether to fail the verification if no named Principal can be created (using one of those 3 claims to get the name), so I'd like to propose to tackle it separately (possibly with the help of the custom Validator as per your suggestion and some specific property). As such I'd like to suggest just to throw the error at this point

Reply:
I understand. Let me have a look and I'll implement something to always fail if is not able to find a sub claim either on `sub` of the configured property. Let's also check `ups` and `preferred _username`.

Reply:
I'd like to have a different issue opened to have a dedicated property (true by default) that if neither of these 3 claims is available then fail. I'm Ok though with it being done as part of this PR, the only question is how to name this property, `require.name.claim` or something like that ?

Reply:
Ignore the last point now please, summarizing below

Reply:
Ok. Thanks. It is becoming somehow confusing with so many parallel comments :)